app/arena.py

```diff
@@ -0,0 +1,30 @@
+from app.knight import Knight
+
+
+class Arena(Knight):
+    __knights = {}
+
+    @classmethod
+    def fighting(cls, pair: list | tuple) -> None:
+        name1, name2 = pair
+        knight1 = cls.__knights[name1]
+        knight2 = cls.__knights[name2]
+
+        if knight1.hp > 0 and knight2.hp > 0:
+            knight1.hp = max(
+                knight1.hp - (knight2.power - knight1.protection), 0
+            )
+            knight2.hp = max(
+                knight2.hp - (knight1.power - knight2.protection), 0
+            )
+        else:
+            print("Dead knights can't fight!")
```

Comment:
Better to make some function like hit(), where first knight hits second, so it's more obvious, and leave that logic inside knight class, while here we only call knight_1.hit(knight(2) and knight_2.hit*knight(1) so we can only see the result.
Also if this message is looking for unexpected behavior, better to Raise an error here, so we know, we have problem somewhere else, since we shouldn't let knights with hp <= 0 to fight. Otherwise we just leave unnecessary print in app state.